ocaml/xapi/xapi_stunnel_server.ml

```diff
@@ -181,8 +181,22 @@ let update_certificates ~__context () =
   | exception e ->
       error "Failed to update host certificates: %s" (Printexc.to_string e)
 
-let restart ~__context ~accept =
+let restart ~__context ipv6_enabled =
   try
+    let accept =
+      let management_enabled =
+        Xapi_inventory.lookup Xapi_inventory._management_interface <> ""
+      in
+      match (management_enabled, ipv6_enabled) with
+      | true, true ->
+          ":::"
+      | true, false ->
+          ""
+      | false, true ->
+          "::1:"
+      | false, false ->
+          "127.0.0.1:"
```

Comment:
Can't we simplify this and always use 127.0.0.1?
Looking at xapi_mgmt_iface.ml in next_server_mode it always listens at least on 127.0.0.1 (and that'll work regardless of whether we also listen to IPv6 or not).

Reply:
This controls what stunnel listens on. To accept external connections, it needs to listen on the actual IP addresses of the host.